authors/apps/profiles/serializers.py

```diff
@@ -0,0 +1,35 @@
+from rest_framework import serializers
+from .models import UserProfile
+
+
+class UserProfileSerializer(serializers.ModelSerializer):
+    """
+    Class to serialize user profile data
+    """
+    profile_image = serializers.ImageField(default='No image')
+    gender = serializers.RegexField(
+        regex='^(N|F|M)$',
+        write_only=True,
+        error_messages={
+            'required': 'Password field required',
+            'invalid': 'Please enter M if you are male, F if you are female or N if you do not want to disclose '
+        }
+    )
+
+    class Meta:
+        model = UserProfile
+        fields = ('first_name', 'last_name', 'gender',
+                  'bio', 'profile_image', 'updated_at')
+        read_only_fields = ('updated_at',)
+
+    def validate(self, data):
+        first_name = data.get('first_name', None)
+        last_name = data.get('last_name', None)
+        email = data.get('email', None)
+        bio = data.get('bio', None)
+        gender = data.get('gender', None)
+        if first_name is None:
+            raise serializers.ValidationError('first name was not provided')
+        if first_name is None:
```

Comment:
Shouldn't this be last_name? But come to think of it, do we really need to overwrite the validate function? Our model fields should handle this already. However, for someone whose Profile was created automatically after registration as a user, should he not be allowed to fill in any of the data when he wants?

Reply:
I have removed the validator for first name and last name